Add formatting using rustfmt

Uxio0 opened the pull request:
- Code formatting will be checked before compilation on _travis_. If code is not formatted, _travis_ will stop there.
- A PR template with the code formatting requisite has been added.
- README was updated with the `rustfmt` instructions.
- Code can be formatted using `cargo fmt --all` or configuring it in the IDEs you are using https://github.com/rust-lang/rustfmt#running-rustfmt-from-your-editor
- All the code has been formatted, so this commit should be merged asap.
- Closes #157

Uxio0 commented:
Coverage decreased just reformatting... I guess the coverage tool takes every line as an instruction...

rmeissner commented on src/models/converters/transactions/tests/mod.rs:
```diff
@@ -1,8 +1,8 @@
-pub(super) mod check_sender_or_receiver;
-pub(super) mod transfer_type_checks;
\ No newline at end of file
+mod summary;
```
Wow this is weird, that the mod visibility has no impact on sorting ... would look nicer if they did :P

rmeissner reviewed (approved):
Having a pre-commit hook would probably be nice :)

Uxio0 commented:
> Having a pre-commit hook would probably be nice :)

I added one script and updated the Readme, but as you know everyone will have to install it. It seems to be working well for me
